### PR TITLE
Fallback to an global comment if the file does not exist in the revision

### PR DESCRIPTION
**Pull request by betterphp**
If the file being reported on does not exist in the revision then a exception is shown in the Jenkins log and a comment is added listing the number of issues. This may be confusing as a comment may be shown that there are more issue than there are comments.

This adds a fallback for the case described to post a global comment with the file name and line number if it's not found in the revision.

### CODE REVIEW

**Comment by nthienan** [comment]
I noticed this issue but have no time to fix it. Thank you for your PR!

File: src/main/kotlin/io/nthienan/phdiff/PhabricatorDifferentialPostJob.kt [hunk 1]
```diff
@@ -45,28 +45,32 @@ class PhabricatorDifferentialPostJob(
...
+              globalReportBuilder.add(i)
+              val ic = inlineReportBuilder.issue(i).build()
+              val filePath = i.componentKey().replace(projectKey, "").substring(1)
+              try {
+                differentialClient.postInlineComment(diffID, filePath, i.line()!!, ic)
+                log.debug("Comment $ic has been published")
+              } catch (e: ConduitException) {
+                if (e.message.equals("Requested file doesn't exist in this revision.")) {
+                  val message = "Unmodified file " + filePath + " on line " + i.line() + "\n\n" + ic
+                  differentialClient.postComment(diff.revisionId, message)
```
Comment (nthienan) [inline src/main/kotlin/io/nthienan/phdiff/PhabricatorDifferentialPostJob.kt:66]: should modify `postComment()` to not attach inline comments in this case

File: src/main/kotlin/io/nthienan/phdiff/PhabricatorDifferentialPostJob.kt [hunk 2]
```diff
@@ -45,28 +45,32 @@ class PhabricatorDifferentialPostJob(
...
+            run {
+              globalReportBuilder.add(i)
+              val ic = inlineReportBuilder.issue(i).build()
+              val filePath = i.componentKey().replace(projectKey, "").substring(1)
+              try {
+                differentialClient.postInlineComment(diffID, filePath, i.line()!!, ic)
+                log.debug("Comment $ic has been published")
+              } catch (e: ConduitException) {
+                if (e.message.equals("Requested file doesn't exist in this revision.")) {
+                  val message = "Unmodified file " + filePath + " on line " + i.line() + "\n\n" + ic
```
Comment (nthienan) [inline src/main/kotlin/io/nthienan/phdiff/PhabricatorDifferentialPostJob.kt:65]: can use string template

**Review by nthienan — APPROVED** [review]

**Comment by betterphp** [comment]
Awesome, thanks :)

**Pull request merged**